sabnzbd/assembler.py

```diff
@@ -326,7 +326,7 @@ def rar_contains_unwanted_file(filepath):
     # returns False if no unwanted extensions are found in the rar file
     # returns name of file if unwanted extension is found in the rar file
     unwanted = None
-    if is_rarfile(filepath):
+    if is_rarfile(filepath) and cfg.unwanted_extensions():
```

Comment:
You may want to change the order of this test.
Testing the option is cheaper than checking the content.